earthfile2llb/interpreter.go

```diff
@@ -1398,13 +1398,13 @@ func (i *Interpreter) handleEnv(ctx context.Context, cmd spec.Command) error {
 	return nil
 }
 
-var errInvalidSyntax = errors.New("invalid syntax")
-var errRequiredArgHasDefault = errors.New("required ARG cannot have a default value")
-var errGlobalArgNotInBase = errors.New("global ARG can only be set in the base target")
+var ErrInvalidSyntax = errors.New("invalid syntax")
+var ErrRequiredArgHasDefault = errors.New("required ARG cannot have a default value")
+var ErrGlobalArgNotInBase = errors.New("global ARG can only be set in the base target")
 
-// parseArgArgs parses the ARG command's arguments
+// ParseArgArgs parses the ARG command's arguments
 // and returns the argOpts, key, value (or nil if missing), or error
-func parseArgArgs(ctx context.Context, cmd spec.Command, isBaseTarget bool, explicitGlobalFeature bool) (commandflag.ArgOpts, string, *string, error) {
+func ParseArgArgs(ctx context.Context, cmd spec.Command, isBaseTarget bool, explicitGlobalFeature bool) (commandflag.ArgOpts, string, *string, error) {
```

Comment:
Exported and used with auto-skip code.

Reply:
I'm not sure if it's a great idea to have the inputgraph call into the earthfile2llb package; this function seems isolated from the rest of the interpreter, so maybe it can be moved into a new file under `util/flagutil/parse_arg_args.go` (or maybe even rename it to `ParseArgCommandArgs` to make it more clear since argargs is a mouthful).

Reply:
Sounds good. Moving some of these shared funcs to a separate package makes sense.

Reply:
I ended up moving a number of shared functions to `flagutil`. Since some deal with arguments rather than de facto flags, I wonder if we should create a separate module. Conversely, the `flagutil` module already contains a few functions that aren't strictly to do with flags.

Reply:
nice work on the refactor here, it's good to see we were able to make it more clear which parts were used in both inputgraph and earthlfile2llb packages.